book/src/user_manual/examining_welcome_message.md

```diff
@@ -0,0 +1,49 @@
+# Examining a welcome message
+
+When a client is invited to join a group, the application can allow the client to decide whether or not to join the group. In order to determine whether to join the group, the application can inspect information provided in the welcome message, such as who invited them, who else is in the group, what extensions are available, and more. If the application decides not to join the group, data must be cleaned up in the storage provider.
+
+After receiving a `MlsMessageIn` from the delivery service, the first step is to extract the `MlsMessageBodyIn`, and determine whether it is a welcome message.
+```rust,no_run,noplayground
+{{#include ../../../openmls/tests/book_code.rs:not_join_group_welcome}}
+```
+
+At this stage, the welcome's `&[EncryptedGroupSecrets]` can be investigated:
+```rust,no_run,noplayground
+{{#include ../../../openmls/tests/book_code.rs:not_join_group_welcome_inspect}}
+```
+
+The next step is to process the `Welcome`.
+```rust,no_run,noplayground
+{{#include ../../../openmls/tests/book_code.rs:not_join_group_processed_welcome}}
+```
+
+After this, some (unverified) information contained in the `ProcessedWelcome` can be investigated.
+```rust,no_run,noplayground
+{{#include ../../../openmls/tests/book_code.rs:not_join_group_processed_welcome_inspect}}
+```
+
+The next step is to stage the `ProcessedWelcome`. This causes some state to be updated in the `StorageProvider`. 
```

Comment:
The relevant information here for the application developer is that up to this point, nothing is persisted. But when the message is processed, it must be discarded afterwards.